database.go

```diff
@@ -63,6 +64,22 @@ func sameNameOnDB(newBook Book) (unique bool) {
 	}
 }
 
+/* Search a book in database based on ID and returns it if succeed. */
+func searchById(id uuid.UUID) (empty bool, storedBook Book) {
+	sqlStatement := `SELECT id, name, price, inventory FROM bookstable WHERE id=$1;`
+	foundRow := dbObject.QueryRow(sqlStatement, id)
+	var bookToReturn Book
+	switch err := foundRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory); err {
+	case sql.ErrNoRows:
+		return true, Book{}
+	case nil:
+		return false, bookToReturn
+	default:
+		panic(err)
```

Comment:
If you return an `error` you don't need to panic here.
Remember when we talked about the function responsibilities and we understood that the migration shouldn't stop execution? This is the same thing.

Reply:
Ok. I did not understand exactly which function may hold the responsibility of stopping execution. It's only the main function? What is the criteria?

Reply:
Usually its only `main`. You should avoid panics as much as you can. They cause unexpected behavior. Panics should only happen in a situation where there is no recovery.

Maybe we should talk about it.